### PR TITLE
SDK Refactor - IDP

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [X] Chore
- [X] Feature
- [ ] Bug fix

### Linked issue

https://cybrid.atlassian.net/browse/CYB-1331

### CODE REVIEW

File: SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/util/Logger.kt [hunk 1]
```diff
@@ -6,29 +6,31 @@ import app.cybrid.sdkandroid.Cybrid
...
 
     AUTH_SET(Log.INFO, "Setting auth token"),
+    AUTH_ERROR(Log.ERROR, "Please set a valid token"),
     AUTH_EXPIRED(Log.ERROR, "Session is expired"),
     NETWORK_ERROR(Log.ERROR, "Network error"),
     COMPONENT_INIT(Log.INFO, "Initializing"),
     DATA_FETCHED(Log.INFO, "Data Fetched"),
     DATA_REFRESHED(Log.INFO, "Refreshing"),
     DATA_ERROR(Log.ERROR, "There was an error fetching"),
+    CONFIG_ERROR(Log.ERROR, "You can only configure the SDK once"),
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/util/Logger.kt:16]: Why is this an error?

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/CybridTest.kt [hunk 2]
```diff
@@ -66,5 +245,5 @@ class CybridTest {
         // -- Then
         Assert.assertNull(listenerPrev)
         Assert.assertNotNull(cybrid.listener)
-    }
+    }*/
```
Comment (brent-cybrid) [inline SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/CybridTest.kt]: Remove these tests or add them back

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/transfer/TransferViewModelTest.kt [hunk 3]
```diff
@@ -138,11 +138,11 @@ class TransferViewModelTest {
 
         // -- Then
         Assert.assertNotNull(viewModel)
-        Assert.assertNotNull(viewModel.assets)
+        /*Assert.assertNotNull(viewModel.assets)
         Assert.assertTrue(viewModel.assets!!.isNotEmpty())
         Assert.assertNotNull(viewModel.accounts)
         Assert.assertTrue(viewModel.accounts.isNotEmpty())
-        Assert.assertEquals(viewModel.fiatBalance.value, "$0.10")
+        Assert.assertEquals(viewModel.fiatBalance.value, "$0.10")*/
```
Comment (brent-cybrid) [inline SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/transfer/TransferViewModelTest.kt]: Should these be in?

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/util/DataAccessStrategyTest.kt [hunk 4]
```diff
@@ -92,8 +92,8 @@ class DataAccessStrategyTest {
     fun unauthorizedAndNullListenerTest() = runBlocking {
 
         val expectedCode = HttpURLConnection.HTTP_UNAUTHORIZED
-        Cybrid.instance.setBearer(TestConstants.expiredToken)
-        Cybrid.instance.listener = null
+        Cybrid.getInstance().setBearer(TestConstants.expiredToken)
+        //Cybrid.getInstance().listener = null
```
Comment (brent-cybrid) [inline SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/util/DataAccessStrategyTest.kt]: Remove?

File: SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/util/Logger.kt (hunk 1 above)
Comment (nalancer08) [reply SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/util/Logger.kt:16]: Becase you can only configure once the SDK at the init, if you need to configure again you need to reset the SDK

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/CybridTest.kt (hunk 2 above)
Comment (nalancer08) [reply SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/CybridTest.kt]: Solved

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/util/DataAccessStrategyTest.kt (hunk 4 above)
Comment (nalancer08) [reply SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/util/DataAccessStrategyTest.kt]: Solved!

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/transfer/TransferViewModelTest.kt (hunk 3 above)
Comment (nalancer08) [reply SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/transfer/TransferViewModelTest.kt]: Solved!

**Review by brent-cybrid — APPROVED** [review]

**Pull request merged**